jax/_src/sharding_impls.py

```diff
@@ -1054,10 +1053,6 @@ def prepare_axis_resources(axis_resources,
       if isinstance(entry, PmapSharding):
         raise ValueError(f'One of {what} got sharding {entry} which is not '
                          'allowed.')
-      if xla_extension_version < 270:
-        if not isinstance(entry, XLACompatibleSharding):
```

Comment:
You probably need to delete the XLACompatibleSharding import from here and pjit.py too?